tests/integration/security/egress_gateway_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,181 @@
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/tests/integration/security/util"
+)
+
+const (
+	// should be templated into YAML, and made interchangeable with other sites
+	externalURL      = "http://edition.cnn.com/politics"
+	externalReqCount = 2
+	egressName       = "istio-egressgateway"
+	// paths to test configs
+	simpleTLSGatewayConfig          = "testdata/gateway-tls-origination.yaml"
+	simpleTLSDestinationRuleConfig  = "testdata/destination-rule-tls-origination.yaml"
+	disableTLSDestinationRuleConfig = "testdata/destination-rule-no-tls-origination.yaml"
+)
+
+// TestEgressGatewayTls brings up an SDS enabled cluster and will ensure that the TLS origination at
+// egress gateway allows secure communication between the egress gateway and external workload.
+// This test brings up an egress gateway to originate TLS connection. The test will ensure that requests
+// are routed securely through the egress gateway and that the TLS origination happens at the gateway.
+func TestSdsEgressGatewayTls(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			ctx.RequireOrSkip(environment.Kube)
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "egress-gateway-tls-origination",
+				Inject: true,
+			})
+			applySetupConfig(ctx, ns)
+
+			testCases := map[string]struct {
+				destinationRulePath string
+				configPath          string
+				response            string
+			}{
+				"SIMPLE TLS origination from egress gateway succeeds": {
+					destinationRulePath: simpleTLSDestinationRuleConfig,
+					configPath:          simpleTLSGatewayConfig,
+					response:            response.StatusCodeOK,
+				},
+				"No TLS origination from egress gateway returns 503 response": {
+					destinationRulePath: disableTLSDestinationRuleConfig,
+					configPath:          simpleTLSGatewayConfig,
+					response:            response.StatusCodeUnavailable,
+				},
+			}
+
+			for name, tc := range testCases {
+				ctx.NewSubTest(name).
+					Run(func(ctx framework.TestContext) {
+						doOriginationTest(ctx, ns, tc.destinationRulePath, tc.configPath, tc.response)
+					})
+			}
+		})
+}
+
+func doOriginationTest(
+	ctx framework.TestContext, ns namespace.Instance, destinationRulePath, configPath, expectedResp string) {
+	var client echo.Instance
+	echoboot.NewBuilderOrFail(ctx, ctx).
+		With(&client, util.EchoConfig("client", ns, false, nil, p)).
+		BuildOrFail(ctx)
+	ctx.ApplyConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, destinationRulePath))
+	ctx.ApplyConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, configPath))
+	defer ctx.DeleteConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, configPath))
+
+	// give the configuration a moment to kick in
+	time.Sleep(time.Second * 20)
+	pretestReqCount := getEgressRequestCountOrFail(ctx, ns, prom)
```

Comment:
Why do we need metrics here? Metrics are super slow and non granular so should be avoided where possible in tests. We are the ones sending traffic, so we know whether it succeeded or failed and can look at the responses directly (which you already are)

Reply:
I was adopting this from sds_egress test, but we don't need metrics so I will get rid of those.